### PR TITLE
[TASK] Streamline "Advanced routing configuration" chapter

### PR DESCRIPTION
**Pull request by brotkrueml**
Mainly adjust formatting and some wording and add links. Additionally, a TOC is now available.

### CODE REVIEW

**Comment by brotkrueml** [comment]
This is a preparation for https://github.com/TYPO3-Documentation/Changelog-To-Doc/issues/273.

File: Documentation/ApiOverview/Routing/AdvancedRoutingConfiguration.rst [hunk 1]
```diff
@@ -8,427 +8,512 @@
...
+    :file:`config.yaml` site configuration file (located in
+    :file:`config/sites/<yoursite>/config.yaml`).
 
-To give you an overview of what the distinction is, we take a regular page which is available at
+To map GET parameters to routes, a concept called "enhancers and aspects has
+been introduced.
+
+An :ref:`enhancer <routing-advanced-routing-configuration-enhancers>` creates
+variants of a specific page-based route for a specific purpose (e.g., one plugin,
+one :ref:`Extbase <extbase>` plugin) and enhances the existing route path, which
```
Comment (linawolf) [inline Documentation/ApiOverview/Routing/AdvancedRoutingConfiguration.rst:32]: one plugin one Extbase plugin ?  I dont get what you mean here

File: Documentation/ApiOverview/Routing/AdvancedRoutingConfiguration.rst [hunk 2]
```diff
@@ -8,427 +8,512 @@
...
+specific route to a page:
 
 :samp:`https://example.org/path-to/my-page/products/<product-name>`
 
-The suffix `/products/<product-name>` to the base route of the page is added by an enhancer. The placeholder variable
-which is added by the curly braces can then be statically or dynamically resolved or built by an Aspect (more
-commonly known as a Mapper).
+The suffix `/products/<product-name>` to the base route of the page is added by
+an enhancer. The placeholder variable added by the curly braces can then be
+resolved statically or dynamically, or built by an Aspect (more commonly
```
Comment (linawolf) [inline Documentation/ApiOverview/Routing/AdvancedRoutingConfiguration.rst:53]:
```suggestion
resolved statically or dynamically, or built by an Aspect (more commonly
```
```suggestion
resolved statically or dynamically, or built by an aspect (more commonly
```

File: Documentation/ApiOverview/Routing/AdvancedRoutingConfiguration.rst (hunk 1 above)
Comment (brotkrueml) [reply Documentation/ApiOverview/Routing/AdvancedRoutingConfiguration.rst]: A pi-based plugin, an Extbase plugin... use now just Extbase plugin.

**Review by linawolf — APPROVED** [review]

**Pull request merged**